src/usb_hub.rs

```diff
@@ -135,6 +136,47 @@ const PORTS: &[(&str, &str)] = &[
     ),
 ];
 
+// The total current for all ports is limited to 700mA, the per-port current is
+// limited to 500mA.
+// The measurement is not _that_ exact so start warning at 90% utilization.
+pub const MAX_TOTAL_CURRENT: f32 = 0.7;
+pub const MAX_PORT_CURRENT: f32 = 0.5;
+const CURRENT_MARGIN: f32 = 0.9;
+
+#[derive(Serialize, Deserialize, Clone, PartialEq)]
+pub enum OverloadedPort {
+    Total,
+    Port1,
+    Port2,
+    Port3,
+}
+
+impl OverloadedPort {
+    fn from_currents(total: f32, port1: f32, port2: f32, port3: f32) -> Option<Self> {
+        let total_overload = total > MAX_TOTAL_CURRENT * CURRENT_MARGIN;
+        let port1_overload = port1 > MAX_PORT_CURRENT * CURRENT_MARGIN;
+        let port2_overload = port2 > MAX_PORT_CURRENT * CURRENT_MARGIN;
+        let port3_overload = port3 > MAX_PORT_CURRENT * CURRENT_MARGIN;
```

Comment:
Nit: please extract `MAX_PORT_CURRENT * CURRENT_MARGING` and `MAX_TOTAL_CURRENT * CURRENT_MARGING` into constants as well and use them directly in the comparison e.g.:

```rust
if total > MAX_TOTAL_CURRENT_XYZ {
  Some(Self::Total)
// ...
```

Reply:
Done

Reply:
Thanks, I find that nicer. If you want to go all the way you could remove the `let total_overload = xxx` bindings and just do the comparison in the if branches. Do this at your own judgement, it is a nit for me.